### PR TITLE
Cleanup proxyserver

### PR DESCRIPTION
**Pull request by vivintw**
* refractors code
* adds unit tests
* creates a configurable whitelist of hosts in proxylet
* Adds basic framework for metrics, will be completed in ASTRACTL-13550

### CODE REVIEW

**Comment by vivintw** [comment]
### Testing 
#### Register https proxylet
```
vivin@vivin-mac-0 natssync % curl -X POST -H 'Content-Type: application/json' -d '{"authToken":"42","locationID":"client1"}' "http://localhost:8083/bridge-client/1/register" | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    94  100    53  100    41    335    259 --:--:-- --:--:-- --:--:--   594
{
  "locationID": "705f70e7-fbec-4b0a-b854-7ce6eb65cb21"
}
```

#### Validate that curl to a configured target succeeds
```
vivin@vivin-mac-0 natssync % curl --insecure -x http://705f70e7-fbec-4b0a-b854-7ce6eb65cb21:@localhost:8082/ http://testnginx -vvv            
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8082 (#0)
* Proxy auth using Basic with user '705f70e7-fbec-4b0a-b854-7ce6eb65cb21'
> GET http://testnginx/ HTTP/1.1
> Host: testnginx
> Proxy-Authorization: Basic NzA1ZjcwZTctZmJlYy00YjBhLWI4NTQtN2NlNmViNjVjYjIxOg==
> User-Agent: curl/7.64.1
> Accept: */*
> Proxy-Connection: Keep-Alive
> 
< HTTP/1.1 200 OK
< Accept-Ranges: bytes
< Connection: keep-alive
< Content-Length: 3
< Content-Type: text/html
< Date: Wed, 08 Dec 2021 14:48:46 GMT
< Etag: "618c1483-3"
< Last-Modified: Wed, 10 Nov 2021 18:50:43 GMT
< Server: nginx/1.21.4
< 
* Connection #0 to host localhost left intact
ok * Closing connection 0
vivin@vivin-mac-0 natssync %
```

#### validate curl to invalid target fails
```
vivin@vivin-mac-0 natssync % curl --insecure -x http://705f70e7-fbec-4b0a-b854-7ce6eb65cb21:@localhost:8082/ http://testnginx1 -vvv
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 8082 (#0)
* Proxy auth using Basic with user '705f70e7-fbec-4b0a-b854-7ce6eb65cb21'
> GET http://testnginx1/ HTTP/1.1
> Host: testnginx1
> Proxy-Authorization: Basic NzA1ZjcwZTctZmJlYy00YjBhLWI4NTQtN2NlNmViNjVjYjIxOg==
> User-Agent: curl/7.64.1
> Accept: */*
> Proxy-Connection: Keep-Alive
> 
< HTTP/1.1 502 Bad Gateway
< Content-Length: 14
< Content-Type: 
< Date: Wed, 08 Dec 2021 14:48:50 GMT
< 
* Connection #0 to host localhost left intact
invalid target* Closing connection 0
vivin@vivin-mac-0 natssync % 
```

File: Makefile [hunk 1]
```diff
@@ -217,12 +217,12 @@ tagAndPushToDockerHub: tag
 	docker push ${IMAGE_REPO}/natssync-tests:latest
 	docker push ${IMAGE_REPO}/httpproxy-server:${IMAGE_TAG}
 	docker push ${IMAGE_REPO}/httpproxy-server:latest
+#   httpproxy_server is depricated use httpproxy-server instead
```
Comment (raghu-sarangapani) [inline Makefile:220]: these 3 were recently removed, not needed

Comment (vivintw) [reply Makefile:220]: thanks I had trouble during the merge

Comment (vivintw) [reply Makefile:220]: I will replace this with the existing make file from main

**Review by palicena — APPROVED** [review]

**Review by jharrod — APPROVED** [review]
I don't have near enough context to judge the implementation but the code looks great

**Review by coffeefreak101 — APPROVED** [review]

**Review by bmason42 — APPROVED** [review]

**Pull request merged**